lib/Game.js

```diff
@@ -0,0 +1,139 @@
+const RandomGenerator = require('./RandomGenerator');
+const GameModeEnum = require('./GameModeEnum');
+
+const DEFAULT_SPEED = 4000;
+const DEFAULT_LENGTH = 1;
+const TIMEOUT_SLEEP = 3000;
+
+module.exports = class Game {
+
+	constructor(translation, gameMode, channel, players) {
+		this.translation = translation;
+		this.gameMode = gameMode;
+		this.channel = channel;
+		this.players = players;
+
+		this.randomGenerator = new RandomGenerator();
+
+		this.firstStart = true;
+		this.currentPlayers = [];
+		this.currentPlayer = null;
+		this.currentString = null;
+		this.currentSpeed = DEFAULT_SPEED;
+		this.currentLength = DEFAULT_LENGTH;
+		this.lastMessageOfBot = null;
+
+		this.sleepTimeout = null;
+		this.callbacks = {};
+	}
+
+	on(key, callback) {
+		this.callbacks[key] = callback;
+	}
+
+	start() {
+		this.callNext();
+	}
+
+	stop() {
+		if (typeof this.callbacks.stop !== 'undefined') {
+			this.callbacks.stop();
+		}
+	}
+
+	callNext() {
+		const playersAlive = Object.values(this.players);
+
+		// Check if more than one player is alive
+		if (playersAlive.length === 1) {
+			this.channel.send(this.translation.f('game.win', { username: playersAlive[0].username }));
+			this.stop();
+			return;
+		}
+
+		if (playersAlive.length === 0) {
+			this.stop();
+			return;
+		}
+
+		// Check if all players played a round
+		if (this.currentPlayers.length === 0) {
+			this.gameNewRound(playersAlive);
+		}
+
+		this.sleepTimeout = setTimeout(this.startNewRound.bind(this), TIMEOUT_SLEEP);
+	}
+
+	startNewRound() {
+		// Create new message
+		this.currentString = this.randomGenerator.getRandomString(this.currentLength);
+
+		// Get random player & send new message
+		const randomPlayerIndex = Math.floor(Math.random() * this.currentPlayers.length);
+		const randomPlayer = this.currentPlayers.splice(randomPlayerIndex, 1);
```

Comment:
please move this logic to a function 
`this.currentPlayer = this.getRandomPlayer();`

Reply:
The whole function is about getting a new player and send him an message. I could rename it, but creating a new function is stupid.